File: src/main/java/no/novari/flyt/authorization/user/UserPublishingComponent.java
package no.novari.flyt.authorization.user;

import lombok.RequiredArgsConstructor;
import org.springframework.scheduling.annotation.Scheduled;
import org.springframework.stereotype.Component;

@Component
@RequiredArgsConstructor
public class UserPublishingComponent {

    private final UserService userService;

    @Scheduled(
            initialDelayString = "${novari.flyt.authorization.access-control.sync-schedule.initial-delay}",
            fixedDelayString = "${novari.flyt.authorization.access-control.sync-schedule.fixed-delay}"
    )
    public void publishUsers() {
        userService.publishUsers();
    }

}

File: src/main/kotlin/no/novari/flyt/authorization/user/UserPublishingComponent.kt
package no.novari.flyt.authorization.user

import org.springframework.scheduling.annotation.Scheduled
import org.springframework.stereotype.Component

@Component
class UserPublishingComponent(
    private val userService: UserService,
) {
    @Scheduled(
        initialDelayString = "\${novari.flyt.authorization.access-control.sync-schedule.initial-delay}",
        fixedDelayString = "\${novari.flyt.authorization.access-control.sync-schedule.fixed-delay}",
    )
    fun publishUsers() {
        userService.publishUsers()
    }
}
